### ISSUE TITLE
Logging should default to stderr, not stderr

### ISSUE THREAD

**Issue by stormmq** [openucx/ucx#1585]
ucx is a library; ideally, it shouldn't be doing any logging at all. However, with something as complex, there is the need for some debug information and tracing.

However, by default, it logs to stdout and not stderr, which isn't good for downstream users. Please default logging to stderr.

Whilst looking at the logging code briefly, I also noticed:-
* There is no obvious way to send log output to syslog, ideally with the ability to choose a facility and a log line prefix (eg `ucx:`) so I can filter for problems in monitoring tools (good log messages are easy to parse and decompose; colon delimited without spaces is a nice choice). Logging to files isn't a wise choice for applications that are long-running. I've witnessed self-induced DoS attacks from logging code in the past, and I will again.
* Some code logs directly to stderr without going via the logging framework, making it impossible to suppress or trace, eg `src/ucm/ptmalloc283/sysdeps/pthread/thread-st.h` line 68 or so. Found by simply typing `stderr` into the GitHub search box. Of course, there may be early-phase code before logging has been initiated
* Logging inflicts a slight runtime cost that could be reduced with a compile time switch that sets the maximum log level; then the `unlikely` and its ilk in `log.h` aren't needed
* It would be useful if the various `print_info` functions could also have a variant that sent their output to the current logging set up, making capture of all salient detail straightforward.

**Comment by shamisp** [comment]
I think you should be able to disable the logging with "--disable-logging"

```src/ucm/ptmalloc283/sysdeps/pthread/thread-st.h``` - this is a 3rd party library that we use for malloc-hooks and registration, so I'm not sure what we will do about this @yosefe

**Comment by stormmq** [comment]
Yes, `--disable-logging` disables tracing, but my point is substantially greater...

**Comment by shamisp** [comment]
@stormmq - I understand. I'm just trying to suggest some near term workaround.

**Comment by stormmq** [comment]
I apologise if my comment seemed snippy; it wasn't my intention.

**Comment by samcmill** [comment]
Has there been any progress on this issue?  I hit this in a containerized CI job.  The job redirects stderr to /dev/null, but the UCX log output to stdout caused a false positive.  

The specific warning was the following, which I was able to disable with `UCX_VFS_ENABLE=n`.
```
[1665672240.391434] [3a6d345564b1:567  :f]        vfs_fuse.c:425  UCX  WARN  failed to connect to vfs socket '??????????????????????????????': Invalid argument
[1665672240.395772] [3a6d345564b1:568  :f]        vfs_fuse.c:425  UCX  WARN  failed to connect to vfs socket '??????????????????????????????': Invalid argument
```

FWIW, this was with HPC-X 2.12.
